src/utils/generator/ReplicaFiller.cpp

```diff
@@ -144,79 +144,90 @@ void ReplicaFiller::setObject(std::shared_ptr<Object> object) { _object = object
 std::shared_ptr<Object> ReplicaFiller::getObject() { return _object; }
 
 void ReplicaFiller::readXML(XMLfileUnits& xmlconfig) {
-	if(xmlconfig.changecurrentnode("input")) {
-		std::string inputPluginName;
-		xmlconfig.getNodeValue("@type", inputPluginName);
-		if(inputPluginName != "BinaryReader") {
-			global_log->error() << "ReplicaFiller only works with inputPlugins: BinaryReader at the moment" << endl;
-			Simulation::exit(1);
-		}
-		setInputReader(std::make_shared<BinaryReader>());
-		_inputReader->readXML(xmlconfig);
-		if(_inputReader == nullptr) {
-			global_log->error() << "Could not create input reader " << inputPluginName << endl;
-			Simulation::exit(1);
-		}
-		xmlconfig.changecurrentnode("..");
-	} else {
-		global_log->error() << "Input reader for original not specified." << endl;
-		Simulation::exit(1);
-	}
-	if(xmlconfig.changecurrentnode("origin")) {
-		Coordinate3D origin;
-		origin.readXML(xmlconfig);
-		origin.get(_origin);
-		xmlconfig.changecurrentnode("..");
-	}
-	global_log->info() << "Base point for the replication: [" << _origin[0] << "," << _origin[1] << "," << _origin[2]
-					   << "]" << endl;
-
-	unsigned int componentid = 0;
-	_componentid = 0;
-	if(xmlconfig.getNodeValue("componentid", componentid))
-		_componentid = componentid;
+    if (xmlconfig.changecurrentnode("input")) {
+        std::string inputPluginName;
+        xmlconfig.getNodeValue("@type", inputPluginName);
+        if (inputPluginName != "BinaryReader") {
+            global_log->error() << "[ReplicaFiller] ReplicaFiller only works with inputPlugins: BinaryReader at the moment" << endl;
+            Simulation::exit(1);
+        }
+        setInputReader(std::make_shared<BinaryReader>());
+        _inputReader->readXML(xmlconfig);
+        if (_inputReader == nullptr) {
+            global_log->error() << "[ReplicaFiller] Could not create input reader " << inputPluginName << endl;
+            Simulation::exit(1);
+        }
+        xmlconfig.changecurrentnode("..");
+    } else {
+        global_log->error() << "[ReplicaFiller] Input reader for original not specified." << endl;
+        Simulation::exit(1);
+    }
+    if (xmlconfig.changecurrentnode("origin")) {
+        Coordinate3D origin;
+        origin.readXML(xmlconfig);
+        origin.get(_origin);
+        xmlconfig.changecurrentnode("..");
+    }
+    global_log->info() << "[ReplicaFiller] Base point for the replication: ["
+                       << _origin[0] << "," << _origin[1] << "," << _origin[2]
+                       << "]" << endl;
+
+    unsigned int componentid = 0;
+    if (xmlconfig.getNodeValue("componentid", componentid)) {
```

Comment:
add a comment on what this `if` does. Suggestion:
```suggestion
	// if the XML defines a new component ID for the replication use it.
	// Otherwise, keep the value that is defined in the phasespace file.
	if (xmlconfig.getNodeValue("componentid", componentid)) {
```